HopsanGenerator/templates/exe_utilities.cpp

```diff
@@ -0,0 +1,488 @@
+        parName.insert(0,systemName+"$");
+    }
+
+    //Convert to short variable names (i.e. "Value" => y")
+    if(endsWith(parName, "#y")) {
+        parName.erase(parName.length()-2,2);
+        parName.append("#Value");
+    };
+
+    std::vector<std::string> vec2, nameVec, syshierarcy;
+    string componentName, parameterName;
+    splitStringOnDelimiter(parName, '$', vec2);
+
+    // Last of vec2 will contain the rest of the name filed (comp and param name)
+    int i;
+    for (i=0; i<int(vec2.size())-1; ++i) {
+        syshierarcy.push_back(vec2[i]);
+    }
+
+    // Split last name part into comp and param name
+    splitStringOnDelimiter(vec2[i], '#', nameVec);
+    if (nameVec.size() == 2 || nameVec.size() == 3 ) {
+        if (nameVec.size() == 2) {
+            componentName = nameVec[0];
+            parameterName = nameVec[1];
+        }
+        if (nameVec.size() == 3) {
+            // Set component name and reset the parameter (startvalue) name
+            componentName = nameVec[0];
+            parameterName = nameVec[1]+"#"+nameVec[2];
+        }
+
+        // Dig down subsystem hiearchy
+        ComponentSystem *pParentSys = pSystem;
+        for (size_t s=1; s<syshierarcy.size(); ++s) {
+            //! @todo what about first level (0), should we check that name is ok
+            ComponentSystem *pSubSys = pParentSys->getSubComponentSystem(syshierarcy[s].c_str());
+            if (!pSubSys) {
+                printErrorMessage("Subsystem: "+syshierarcy[s]+" could not be found in parent system: "+pParentSys->getName().c_str());
+                return false;
+            }
+            else {
+                pParentSys = pSubSys;
+            }
+        }
+
+        // Set the parameter value if component is found
+        if (pParentSys) {
+            Component *pComp = 0;
+            // If syshierarcy is empty then we are setting a parameter in the top-level system
+            if (syshierarcy.empty()) {
+                pComp = pParentSys;
+            }
+            else {
+                pComp = pParentSys->getSubComponent(componentName.c_str());
+            }
+
+            if (pComp) {
+                bool ok = pComp->setParameterValue(parameterName.c_str(), rParValue.c_str());
+                std::cout << "Setting parameter: " << pComp->getName().c_str() << "." << parameterName << " = " << rParValue << "\n";
+                if (!ok) {
+                    std::cout << "Failed!\n";
+                    return false;
+                }
+            }
+            else {
+                printErrorMessage("No component: " + componentName + " in system: " + pParentSys->getName().c_str() );
+                return false;
+            }
+        }
+    }
+    else {
+        std::cout << "Unknown error!\n";
+        return false;
+    }
+
+    return true;
+}
+
+void printHelpText(ComponentSystem *pSystem)
+{
+    std::cout << "This is a stand-alone executable Hopsan model.\n";
```

Comment:
Why not use TCLAP like in the CLI application?

Reply:
I want to minimize external dependencies. It should only support very basic arguments anyway.